tfexec/terraform.go

```diff
@@ -129,6 +133,18 @@ func (tf *Terraform) SetAppendUserAgent(ua string) {
 	tf.appendUserAgent = ua
 }
 
+// SetDisablePluginTls sets the TF_DISABLE_PLUGIN_TLS environment variable for
+// Terraform CLI execution.
+func (tf *Terraform) SetDisablePluginTls(disabled bool) {
```

Comment:
You can probably just make this return an error for compatibility mismatch that people can just ignore if they want.